packages/replay-next/src/suspense/LogPointAnalysisCache.ts

```diff
@@ -0,0 +1,274 @@
+import {
+  ExecutionPoint,
+  Location,
+  PauseData,
+  PauseId,
+  PointDescription,
+  PointRange,
+  TimeStampedPoint,
+} from "@replayio/protocol";
+import jsTokens from "js-tokens";
+
+import {
+  AnalysisInput,
+  AnalysisResultWrapper,
+  SendCommand,
+  getFunctionBody,
+} from "protocol/evaluation-utils";
+import { ReplayClientInterface } from "shared/client/types";
+
+import { createFetchAsyncFromFetchSuspense } from "../utils/suspense";
+import { RemoteAnalysisResult, getAnalysisCache } from "./AnalysisCache";
+import { createGenericCache } from "./createGenericCache";
+
+type Value = any;
+
+export type AnalysisResult = {
+  executionPoint: ExecutionPoint;
+  isRemote: boolean;
+  pauseId: PauseId | null;
+  time: number;
+  values: Value[];
+};
+
+export type GetAnalysisResult = (timeStampedPoint: TimeStampedPoint) => AnalysisResult | null;
+
+export function getLogPointAnalysisResultSuspense(
+  client: ReplayClientInterface,
+  range: PointRange,
+  point: TimeStampedPoint,
+  location: Location,
+  code: string,
+  condition: string | null
+): AnalysisResult {
+  if (!condition && canRunLocalAnalysis(code)) {
```

Comment:
(I know this code existed before but as I'm reading it again after not looking for a while–) it might be nice to update `canRunLocalAnalysis` to accept both `code` and `condition` 😄  since it is kind of silly, semantically, to have a conditional for a local analysis.

Reply:
But a `condition` with a constant value doesn't make much sense, so I don't think we need to optimize for that.

Reply:
I agree. I'm not saying the `canRunLocalAnalysis` function should _check_ if the condition is local, just that it might be more semantically accurate if we moved the check there, e.g.:

```ts
export function canRunLocalAnalysis(code: string, condition: string | null): boolean {
  if (condition) {
    return false;
  }

  // ...
```